simphony/testing/abc_check_engine.py

```diff
@@ -177,15 +177,17 @@ def test_delete_dataset(self):
             name = "test_" + str(i)
             engine.add_dataset(self.create_dataset(name=name))
 
+        datasets_and_names = [(ds, ds.name) for ds in engine.iter_datasets()]
+
         # delete each of the datasets
-        for ds in engine.iter_datasets():
-            engine.remove_dataset(ds.name)
+        for (dataset, name) in datasets_and_names:
+            engine.remove_dataset(name)
             # test that we can't get deleted datasets
             with self.assertRaises(ValueError):
-                engine.get_dataset(ds.name)
+                engine.get_dataset(name)
             # test that we can't use the deleted datasets
             with self.assertRaises(Exception):
-                self.compare_dataset(ds, ds)
+                self.compare_dataset(dataset, dataset)
```

Comment:
Yeah, JYULB wrapper fails at the moment for this test.   I would suggest opening up an issue in simphony-jyulb to follow it and override the method in your subclass of LatticeEngineCheck.

We want something like the following to happen:

```
foo_proxy_lattice = jyulb_engine.get_dataset("foo")
for node in foo_proxy_lattice.iter_nodes():
     print node.data[CUBA.VELOCITY]
jyulb_engine.remove_dataset("foo")
print(type(foo_proxy_lattice))  # prints <class 'jyulb....ProxyLattice''>
for node in foo_proxy_lattice.iter_nodes():  # this should raise an exception
```